### ISSUE TITLE
Swig generation for R fails if function prototype uses the keyword  "result"

### ISSUE THREAD

**Issue by purplefishies** [swig/swig#644]
If you have a function definition as follows

###  /* foo.h */
#ifndef FOO_H
float *foo( int *result );
#endif

and 

### /* foo.c */
#include  "foo.h"
float *foo( int *res )
{
    static float value = 100;
    float *retval = &value;
    int result = 43;
    printf("Going to return a result\n");
    *res = 42;
    printf("Other result: %d\n", result );
    returl retval;
}

And then run Swig to generate a wrapper, I get a wrapper that has the following entry

SWIGEXPORT SEXP
R_swig_foo ( SEXP result)
{
**float *result = 0 ;**   /* This will cause a compiler error */
 int *arg1 = (int *) 0 ;
  unsigned int r_nprotect = 0;
  SEXP r_ans = R_NilValue ;
  VMAXTYPE r_vmax = vmaxget() ;
...

What you get is that result is used twice.  This is different than if you run python generators where the name is always called   ***SWIGUNUSEDPARM(self)*  and  **args**

## How to reproduce 

Using these files
[RSwigFail.zip](https://github.com/swig/swig/files/200154/RSwigFail.zip)

```bash
unzip RSwigFail.zip
cd RSwigFail
swig -r 
R CMD SHLIB -o AIOUSB.so Foo_wrap.c foo.c
```

**Comment by ojwb** [comment]
@joequant Did you see this report?  It seems there's a fixed name `result` which should probably be `swig_result` or similar, or else the parameter name shouldn't just use the name from the function being wrapped.  The latter would avoid similar issues with `arg1`, etc, but perhaps there's a reason to want to use the same parameter name here.

**Comment by joequant** [comment]
Yup.  Trying to come up with a general fix, but I might just hack it by
using a internal name like _swig_result.  Will fix this weekend.

**Comment by ojwb** [comment]
I've looked at this a bit, and I think actually the answer is to not use the C/C++ parameter names for the wrapper function parameter names.  There are many potential collisions, not just `result`.  For example `arg1`, `arg2`, etc (which could be `swig_` prefixed) but also collisions with the R API which can't be avoided except by renaming the parameter - consider `Examples/r/simple` with this simple change - a totally valid C API to be wrapping:

```
--- a/Examples/r/simple/example.i
+++ b/Examples/r/simple/example.i
@@ -2,6 +2,6 @@
 %module example
 
 %inline %{
-extern int    gcd(int x, int y);
+extern int    gcd(int vmaxset, int Rf_error);
 extern double Foo;
 %}
```

But:

```
swig/Examples/r/simple$ make -s
example_wrap.c: In function ‘R_swig_gcd’:
example_wrap.c:1333:3: error: called object ‘vmaxset’ is not a function or function pointer
 1333 |   vmaxset(r_vmax);
      |   ^~~~~~~
example_wrap.c:1320:19: note: declared here
 1320 | R_swig_gcd ( SEXP vmaxset, SEXP Rf_error, SEXP s_swig_copy)
      |              ~~~~~^~~~~~~
example_wrap.c:1338:3: error: called object ‘Rf_error’ is not a function or function pointer
 1338 |   Rf_error("%s %s", SWIG_ErrorType(SWIG_lasterror_code), SWIG_lasterror_msg);
      |   ^~~~~~~~
example_wrap.c:1320:33: note: declared here
 1320 | R_swig_gcd ( SEXP vmaxset, SEXP Rf_error, SEXP s_swig_copy)
      |                            ~~~~~^~~~~~~~
make[2]: *** [/usr/lib/R/etc/Makeconf:168: example_wrap.o] Error 1
make[1]: *** [../../Makefile:1315: r] Error 1
make: *** [Makefile:12: build] Error 2
```

For reasons like these other SWIG target languages generally don't use the C/C++ parameter names in the equivalent situations.
